### PR TITLE
BOM-67

### PR DESCRIPTION
**Pull request by mraarif**
### Checklist
- [x] Modernize code base
- [x] Isort
- [ ] Get Tests running on python 2 and python 3
- [ ] Update classifiers in [setup.py|http://setup.py]
- [ ] Add openedx.yaml
- [ ] Update the version number to 2.2.9.edx-2
- [ ] Tag the new version.
- [ ] Update [github.in|http://github.in] to point to the new version of django-oauth-plus

### CODE REVIEW

File: oauth_provider/runtests/manage.py [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+import os
+import sys
 
-try:
-    import settings # Assumed to be in the same directory.
-except ImportError:
-    import sys
-    sys.stderr.write("Error: Can't find the file 'settings.py' in the directory containing %r. It appears you've customized things.\nYou'll have to run django-admin.py, passing it your settings module.\n(If the file settings.py does indeed exist, it's causing an ImportError somehow.)\n" % __file__)
-    sys.exit(1)
+# from django.core.management import execute_manager
```
Comment (awais786) [inline oauth_provider/runtests/manage.py:8]: @mraarif why you comment out all this ?

File: oauth_provider/tests/protocol.py [hunk 2]
```diff
@@ -1,409 +1,409 @@
...
+    def test_401_for_wrong_scope(self):
+        # If you try to access a resource with a wrong scope, it will return an error
+        parameters = self._make_request_token_parameters()
+        parameters['scope'] = 'videos'
+        parameters['oauth_nonce'] = 'requestnoncevideos'
+
+        response = self.c.get("/oauth/request_token/", parameters)
+
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response.content.decode('utf-8'), 'Scope does not exist.')
```
Comment (awais786) [inline oauth_provider/tests/protocol.py:102]: @mraarif ?

File: setup.py [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+from oauth_provider import __version__ as version
+
+
 def strip_comments(l):
     return l.split('#', 1)[0].strip()
 
 def reqs(*f):
-    return list(filter(None, [strip_comments(l) for l in open(
-        os.path.join(os.getcwd(), *f)).readlines()]))
+    return list([_f for _f in [strip_comments(l) for l in open(
```
Comment (awais786) [inline setup.py:14]: @mraarif is this automatic conversion ?

File: oauth_provider/tests/protocol.py (hunk 2 above)
Comment (mraarif) [reply oauth_provider/tests/protocol.py:102]: for `python 3` the `bytes` output caused the test failure so converted it to `str`

File: oauth_provider/runtests/manage.py (hunk 1 above)
Comment (mraarif) [reply oauth_provider/runtests/manage.py:8]: this should be removed, deprecated way of how django's `manage.py` used to work, it caused runtime errors on newer versions of django

File: oauth_provider/migrations/0001_initial.py [hunk 4]
```diff
@@ -66,9 +67,9 @@ class Migration(migrations.Migration):
                 ('verifier', models.CharField(max_length=10)),
                 ('callback', models.CharField(max_length=2083, null=True, blank=True)),
                 ('callback_confirmed', models.BooleanField(default=False)),
-                ('consumer', models.ForeignKey(to='oauth_provider.Consumer')),
-                ('scope', models.ForeignKey(blank=True, to='oauth_provider.Scope', null=True)),
-                ('user', models.ForeignKey(related_name=b'tokens', blank=True, to=settings.AUTH_USER_MODEL, null=True)),
+                ('consumer', models.ForeignKey(to='oauth_provider.Consumer',on_delete=models.DO_NOTHING)),
```
Comment (mraarif) [inline oauth_provider/migrations/0001_initial.py:70]: while testing, on newer versions of `django` the `on_delete` attribute is required, [it was causing runtime errors]

File: setup.py (hunk 3 above)
Comment (mraarif) [reply setup.py:14]: yes

File: oauth_provider/migrations/0001_initial.py (hunk 4 above)
Comment (awais786) [reply oauth_provider/migrations/0001_initial.py:70]: @feanil any concerns on this ?

**Pull request closed without merging**